packages/client-sdk-web/src/internal/data-client.ts

```diff
@@ -1782,6 +1807,767 @@ export class DataClient<
+    return await new Promise(resolve => {
+      this.clientWrapper.sortedSetGetScore(
+        request,
+        {
+          ...this.authHeaders,
+          ...metadata,
+        },
+        (err, resp) => {
+          if (resp?.getMissing()) {
+            resolve(new CacheSortedSetGetScores.Miss());
+          } else if (resp?.getFound()) {
+            const elements = resp
+              .getFound()
+              ?.getElementsList()
+              .map(ele => {
+                const result = this.convertECacheResult(ele.getResult());
+                return new _SortedSetGetScoreResponsePart(
+                  result,
+                  ele.getScore()
+                );
+              });
+            if (elements) {
+              resolve(
+                new CacheSortedSetGetScores.Hit(
+                  elements,
+                  this.convertArrayToUint8(values.map(val => atob(val)))
+                )
+              );
+            } else {
+              resolve(
+                new CacheSortedSetFetch.Error(
+                  new UnknownError('Unknown sorted set fetch hit response type')
+                )
+              );
+            }
+          } else {
+            resolve(
+              new CacheSortedSetGetScores.Error(cacheServiceErrorMapper(err))
+            );
+          }
+        }
+      );
+    });
+  }
+
+  public async sortedSetGetRank(
+    cacheName: string,
+    sortedSetName: string,
+    value: string | Uint8Array
+  ): Promise<CacheSortedSetGetRank.Response> {
+    try {
+      validateCacheName(cacheName);
+      validateSortedSetName(sortedSetName);
+    } catch (err) {
+      return new CacheSortedSetGetRank.Error(normalizeSdkError(err as Error));
+    }
+
+    this.logger.trace(
+      "Issuing 'sortedSetGetRank' request; value: %s",
+      truncateString(value.toString())
+    );
+
+    const result = await this.sendSortedSetGetRank(
+      cacheName,
+      this.convertToB64String(sortedSetName),
+      this.convertToB64String(value)
+    );
+
+    this.logger.trace(
+      "'sortedSetGetRank' request result: %s",
+      truncateString(result.toString())
+    );
+    return result;
+  }
+
+  private async sendSortedSetGetRank(
+    cacheName: string,
+    sortedSetName: string,
+    value: string
+  ): Promise<CacheSortedSetGetRank.Response> {
+    const request = new _SortedSetGetRankRequest();
+    request.setSetName(sortedSetName);
+    request.setValue(value);
+    const metadata = this.createMetadata(cacheName);
+    return await new Promise(resolve => {
+      this.clientWrapper.sortedSetGetRank(
+        request,
+        {
+          ...this.authHeaders,
+          ...metadata,
+        },
+        (err, resp) => {
+          if (resp?.getMissing()) {
+            resolve(new CacheSortedSetGetRank.Miss());
+          } else if (resp?.getElementRank()) {
+            const rank = resp?.getElementRank()?.getRank();
+            if (!rank) {
```

Comment:
This differs from the nodejs client, where the test is for rank being `undefined`. However, in this client, we're getting back a 0 from `resp?.getElementRank()?.getRank()` when the value isn't in the set. Not sure why `getMissing()` isn't making the miss explicit here.

Reply:
cc: @wandaitzuchen @bruuuuuuuce . 

spec is here: https://www.notion.so/momentohq/SDK-API-Methods-Spec-cf12e4db2535405ca2acc3608814b083?pvs=4#fdace86a5cb349df98b37e8f39d91eea

Reply:
Similar many other collection read APIs, `missing` means *the collection* does not exist; `element_rank` is like `found`, which means collection exists. An `element_rank` includes a `result` field, which is a `ECacheResult` specifying whether *the value* is a Hit or Miss. The semantics of `SortedSetGetRank` is pretty much the same as `SortedSetGetScore`, which, when the set exists, returns a `found` that includes a list of elements, each element includes a `result` field. 

See proto def for [SortedSetGetRank](https://github.com/momentohq/client-protos/blob/89d96a2cb369c5690ee73fe3aa6cd6af7f5b9792/proto/cacheclient.proto#L853-L868) and [SortedSetGetRank](https://github.com/momentohq/client-protos/blob/89d96a2cb369c5690ee73fe3aa6cd6af7f5b9792/proto/cacheclient.proto#L788-L805)